x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/ml/change_point_agg.yml

```diff
@@ -211,3 +261,137 @@ setup:
             }
           }
   - is_true: aggregations.change_point.type.indeterminable
+---
+"Test change_point with terms":
+  - do:
+      search:
+        index: store
+        size: 0
+        body: >
+          {
+            "aggs": {
+              "terms": {
+                 "terms": {
+                   "field": "kind"
+                 },
+                 "aggs": {
+                    "date": {
+                      "date_histogram": {
+                        "field": "time",
+                        "fixed_interval": "1s"
+                      },
+                      "aggs": {
+                        "min": {
+                          "min": {
+                            "field": "cost"
+                          }
+                        }
+                      }
+                    },
+                    "change_point": {
+                      "change_point": {
+                        "buckets_path": "date>min"
+                      }
+                    }
+                 }
+              }
+            }
+          }
+  - is_true: aggregations.terms.buckets.0.change_point.type.trend_change
+  - is_true: aggregations.terms.buckets.0.change_point.type.trend_change.p_value
+  - is_true: aggregations.terms.buckets.0.change_point.type.trend_change.r_value
+  - is_false: aggregations.terms.buckets.1.change_point.type.trend_change
+  - is_false: aggregations.terms.buckets.1.change_point.type.trend_change.p_value
+  - is_false: aggregations.terms.buckets.1.change_point.type.trend_change.r_value
+
+  - do:
+      search:
+        index: store
+        size: 0
+        body: >
+          {
+            "aggs": {
+              "terms": {
+                 "terms": {
+                   "field": "kind"
+                 },
+                 "aggs": {
+                    "date": {
+                      "date_histogram": {
+                        "field": "time",
+                        "fixed_interval": "1s"
+                      },
+                      "aggs": {
+                        "min": {
+                          "min": {
+                            "field": "cost"
+                          }
+                        }
+                      }
+                    },
+                    "change_point": {
+                      "change_point": {
+                        "buckets_path": "date>min"
+                      }
+                    },
+                    "select": {
+                      "bucket_selector": {
+                        "buckets_path": {"p_value": "change_point.p_value"},
+                        "script": "params.p_value < 0.5"
+                      }
+                    }
```

Comment:
Looks great! I assume the following should also work:

```
                    "select": {
                      "bucket_selector": {
                        "buckets_path": {"p_value": "change_point.p_value", "change_point_type": "change_point.type"},
                        "script": "params.p_value < 0.5 && params.change_point_type == \"trend_change\""
                      }
                    }
```

Going to test it in a bit

Reply:
@darnautov bucket_selector doesn't work with strings. It has to point to a numeric. Allowing bucket_selector to filter on strings would be a fundamental change to how that agg works and seems out of scope here :)